src/db/migrations/20250527170707_recreate_gid_proc.js

```diff
@@ -0,0 +1,53 @@
+/*****
+ License
+ --------------
+ Copyright © 2020-2025 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Mojaloop Foundation for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Mojaloop Foundation
+ * Eugen Klymniuk <eugen.klymniuk@infitx.com>
+
+ --------------
+ ******/
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function(knex) {
+  await knex.raw(`DROP PROCEDURE IF EXISTS create_gid`);
+  return knex.raw(`
+    CREATE PROCEDURE create_gid()
+    BEGIN
+        SET @update_id := 0;
+        UPDATE gid
+        SET id = (SELECT @update_id := id + 1)
+        WHERE id > 0
+        LIMIT 1;
```

Comment:
Preferably we should use equal sign here
`WHERE id=1`
and remove the `LIMIT`

Then check the query affected one row with `mysql_affected_rows()` and return error if not.